maas/resource_maas_boot_source_selection_test.go

```diff
@@ -40,9 +39,26 @@ func TestAccResourceMAASBootSourceSelection_basic(t *testing.T) {
 		CheckDestroy: testAccCheckMAASBootSourceSelectionDestroy,
 		ErrorCheck:   func(err error) error { return err },
 		Steps: []resource.TestStep{
+			// Test creation
 			{
-				Config: testAccMAASBootSourceSelection(os, release, arches, subarches, labels),
-				Check:  resource.ComposeAggregateTestCheckFunc(checks...),
+				Config: testAccMAASBootSourceSelection(os, release, arches),
+				Check: resource.ComposeAggregateTestCheckFunc(append(
+					checks,
+					resource.TestCheckResourceAttr("maas_boot_source_selection.test", "arches.#", fmt.Sprintf("%v", len(arches))),
+					resource.TestCheckResourceAttr("maas_boot_source_selection.test", "arches.0", arches[0]),
+				)...,
+				),
+			},
+			// Test update
+			{
+				Config: testAccMAASBootSourceSelection(os, release, updatedArches),
+				Check: resource.ComposeAggregateTestCheckFunc(append(
+					checks,
+					resource.TestCheckResourceAttr("maas_boot_source_selection.test", "arches.#", fmt.Sprintf("%v", len(updatedArches))),
+					resource.TestCheckResourceAttr("maas_boot_source_selection.test", "arches.0", updatedArches[0]),
+					resource.TestCheckResourceAttr("maas_boot_source_selection.test", "arches.1", updatedArches[1]),
```

Comment:
I'm assuming we can guarantee the ordering here?

Reply:
That's a great catch. I updated the tests to tolerate any change in ordering.